idseq_dag/steps/download_accessions.py

```diff
@@ -79,14 +94,26 @@ def download_ref_sequences_from_s3(self, accession_dict, output_reference_fasta,
             t.join()
         if error_flags:
             raise RuntimeError("Error in getting sequences by accession list.")
+
+        for accession, taxinfo in accession_dict.items():
+            accession_out_file = os.path.join(accession_dir, accession)
+            cmd = f"cat {accession_out_file} |head -1"
+            seq_name = subprocess.check_output(
+                    cmd, executable='/bin/bash', shell=True).decode("utf-8").split(" ", 1)[1].replace("\n", "")
+            accession_summary[accession]["name"] = seq_name
```

Comment:
this might actually not be thread-safe though. Also, it's probably easier to get the sequence length here?

Reply:
What would make something not thread-safe? Not sure what I should be thinking about.

Reply:
@yunfangjuan do you want to elaborate? This happens after the threads are joined, no?

Reply:
Writing to the same dict simultaneously

Reply:
To me it looks as if all the download threads are joined in line 94 (https://github.com/chanzuckerberg/idseq-dag/blob/ffef5bf9a6aa7b0284775d9f11a7818e28dc4577/idseq_dag/steps/download_accessions.py#L94) and writing to the dict is single-threaded.

The two places in the code that write to `accession_summary`, namely the methods `download_ref_sequences_from_s3` and `get_sequence_by_accession_from_file`, are called in a mutually exclusive way in the code.

@yunfangjuan Anything I'm missing? Please advise.

Reply:
OK. This case it's actually done after the join so there's only one thread doing it, so it's fine. But I think a better way to deal with this will be to generate another dictionary from the generate_loc_db step to pull out the seq_name and sequence length once for each index update so you won't add time to our current pipeline.